### ISSUE TITLE
[TST] test_computation_output_1 (expand_grid test) fails with dtype difference

### ISSUE THREAD

**Issue by hectormz** [ericmjl/pyjanitor#675]
# Brief Description

Running the test suite on my computer fails with the `test_computation_output_1` test in `test_expand_grid.py`:
```python
____________________________________________________ test_computation_output_1 _____________________________________________________ 

    def test_computation_output_1():
        """Test output if entry contains no dataframes/series"""
        data = {"x": range(1, 4), "y": [1, 2]}
        expected = pd.DataFrame({"x": [1, 1, 2, 2, 3, 3], "y": [1, 2, 1, 2, 1, 2]})
>       assert_frame_equal(expand_grid(others=data), expected)
E       AssertionError: Attributes are different
E       
E       Attribute "dtype" are different
E       [left]:  int32
E       [right]: int64

tests\functions\test_expand_grid.py:229: AssertionError
```
https://github.com/ericmjl/pyjanitor/blob/62ee72fc5ed2fe6d80acc6663d222458252c03de/tests/functions/test_expand_grid.py#L225

For some reason, the manner in which these dataframes are constructed/processed results in different `dtype`s. The strange thing is that none of the other tests in this file are failing, and this is seemingly passing in Azure. 

# System Information

- Operating system: Windows
- Python version: 3.7/3.8
- `numpy`: 1.18.5
- `pandas`: 1.0.4

I have tried this on both 3.8 and 3.7 (which is what our Azure CI runs), and both give the same failure. My guess is that this may be due to the fact that I'm using vanilla Python (not Anaconda like other devs, or what is run on Azure).

I'm curious what others get for these `dtype`s. The easiest fix would be to cast the `dtype`(s) of the expected dataframe, but I'm still perplexed why this happening for this test, and for me in particular. Unless this reveals something requiring a large fix, this could be combined with #674

**Comment by samukweku** [comment]
@hectormz, is this still failing on your end ?

**Comment by hectormz** [comment]
@samukweku I believe I still am. You're using Anaconda + Docker right? I think I was going to setup the pyjanitor env in Anaconda and see if that is different.

**Comment by hectormz** [comment]
Still tracking it down and seeing what can be done. In part it arises with Windows using `int32` by default, and other systems use int64. 

So on Windows,
```python
data = {"x": range(1, 4), "y": [1, 2]}
```
this dictionary, and the processed dataframe are `int32`, while 
```python
expected = pd.DataFrame({"x": [1, 1, 2, 2, 3, 3], "y": [1, 2, 1, 2, 1, 2]})
```
is `int64`.

So not sure if the _fix_ is to adjust the test, or the actual function.

The strange thing is that although a Windows CI machine is used, it must be still getting `int64` from the actual Linux machine.

I've already fuzzed on the details, but I believe I ran this last week on Anaconda, on Windows, and still got this issue.

**Comment by samukweku** [comment]
Thanks for this @hectormz

If the function is to be adjusted, do we have to adjust for integer sizes (I have no idea about that)?

**Comment by hectormz** [comment]
@samukweku Well, I suppose the function is behaving correctly. It is getting an `int32` and returning the same type. 

Perhaps the easiest/best thing to do is actually add `check_dtype` arg:

```python
assert_frame_equal(expand_grid(others=data), expected, check_dtype=False)
```
That clears up the issue. We could add a comment in there about difference in default `int` for Windows etc. 

I was just surprised that this is the first time this has come up. The test to get an idea of what was differing in differing OSes and distributions was:

```python
>>> import numpy as np
>>> a = np.array([1,2,3])
>>> a.dtype
dtype('int32')
```

On Ubuntu I get:
```python
>>> import numpy as np
>>> a = np.array([1,2,3])
>>> a.dtype
dtype('int64')
```

**Comment by samukweku** [comment]
@hectormz this is interesting for me - I will run it on a windows/linux machine and see if i get the same results. Curious to learn from you how you go about your tests, as this relates to infrastructure, which I would love to have an understanding of.

**Comment by ericmjl** [comment]
I don't want to hijack the thread, but @zbarry and @samukweku, could you two send me your Azure, Travis CI and PyPI usernames via email? My current email address is <github-username>[at]hey.com. I need to add your usernames to those systems onto the relevant projects (somehow).

Now returning the thread back to regular programming...

**Comment by hectormz** [comment]
@ericmjl  Jumping on the hey wagon, eh?

@samukweku I'm happy to answer whatever I can. In this case, once I had the failing test, I began to wonder what was going on, and stumbled on a StackOverflow post discussing the differences in default int types in numpy.

From your PR you seem to have testing down. Do you have additional questions about tests or running tests? Anything we should be putting into our docs?

**Comment by samukweku** [comment]
Quite clear @hectormz . Thanks for the explanation.

**Comment by hectormz** [comment]
@samukweku great! Would you be interested in trying out and making a PR with the proposed changes? Otherwise I can do it as well

**Comment by samukweku** [comment]
Sure, I will put in a PR

**Comment by hectormz** [comment]
Closed by #692
